pkg/events/events.go

```diff
@@ -176,7 +178,25 @@ func New(ctx context.Context, name, description string, opts ...Option) Event {
 	return (&definition{name: name, description: description}).New(ctx, opts...)
 }
 
+var errMarshalData = errors.Define("marshal_data", "failed to marshal data")
+
 func marshalData(data any) (anyPB *anypb.Any, err error) {
+	defer func() {
```

Comment:
Let's put a TODO comment and link to the referenced issue. We should remove this. I don't like having these catch-alls in the code base and rather actually panic when something goes wrong. We might oversee other stuff that we need to see.

Reply:
Done.